test/lib/mayaUsd/render/vp2RenderDelegate/testVP2RenderDelegateIsolateSelect.py

```diff
@@ -0,0 +1,160 @@
+#
+# Copyright 2021 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import fixturesUtils
+import imageUtils
+import mayaUtils
+import unittest
+import usdUtils
+import testUtils
+
+from mayaUsd import lib as mayaUsdLib
+from mayaUsd import ufe as mayaUsdUfe
+
+from maya import cmds
+import maya.api.OpenMayaRender as omr
+
+from pxr import Usd
+
+import ufe
+
+import os
+
+
+class testVP2RenderDelegateIsolateSelect(imageUtils.ImageDiffingTestCase):
+    """
+    Tests imaging using the Viewport 2.0 render delegate when using per-instance
+    inherited data on instances.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        inputPath = fixturesUtils.setUpClass(__file__,
+            initializeStandalone=False, loadPlugin=False)
+
+        cls._baselineDir = os.path.join(inputPath,
+            'VP2RenderDelegateIsolateSelectTest', 'baseline')
+
+        cls._testDir = os.path.abspath('.')
+
+    @classmethod
+    def tearDownClass(cls):
+        panel = mayaUtils.activeModelPanel()
+        cmds.modelEditor(panel, edit=True, useDefaultMaterial=False)
+
+    def assertSnapshotClose(self, imageName):
+        cmds.undoInfo(stateWithoutFlush=False) # disable the undo queue during the snapshot
+        baselineImage = os.path.join(self._baselineDir, imageName)
+        snapshotImage = os.path.join(self._testDir, imageName)
+        imageUtils.snapshot(snapshotImage, width=960, height=540)
+        retVal = self.assertImagesClose(baselineImage, snapshotImage)
+        cmds.undoInfo(stateWithoutFlush=True)
+        return retVal
+
+    def testIsolateSelect(self):
+        cmds.file(force=True, new=True)
+        mayaUtils.loadPlugin("mayaUsdPlugin")
+        panel = mayaUtils.activeModelPanel()
+        usdaFile = testUtils.getTestScene("setsCmd", "5prims.usda")
+        proxyDagPath, sphereStage = mayaUtils.createProxyFromFile(usdaFile)
+        usdCube = proxyDagPath + ",/Cube1"
+        usdCylinder = proxyDagPath + ",/Cylinder1"
+        usdCapsule = proxyDagPath + ",/Capsule1"
+        usdCone = proxyDagPath + ",/Cone1"
+        usdXform = proxyDagPath + ",/Xform1"
+
+        cmds.move(-4, -24, 0, "persp")
+        cmds.rotate(90, 0, 0, "persp")
+
+        globalSelection = ufe.GlobalSelection.get()
+        globalSelection.clear()
+        self.assertSnapshotClose('unselected.png')
+
+        # Turn on isolate select for cube
+        cmds.select(usdCube)
+        cmds.isolateSelect("modelPanel4", state=1)
```

Comment:
Can this function use the result of `mayaUtils.activeModelPanel()` as first parameter? Would it make the test more robust?